pandas/tests/test_categorical.py

```diff
@@ -4122,8 +4123,9 @@ def f():
         c[0] = np.nan
         df = pd.DataFrame({"cats": c, "vals": [1, 2, 3]})
 
-        cat_exp = Categorical(["a", "b", "a"], categories=["a", "b", np.nan])
```

Comment:
Again, the test is expecting a `Categorical` with `NaN` in the categories AFAICT.